lib/service/hive_service.dart

```diff
@@ -0,0 +1,77 @@
+// hiveService abstract class
+import 'package:autonomy_flutter/model/eth_pending_tx_amount.dart';
+import 'package:autonomy_flutter/util/log.dart';
+import 'package:hive/hive.dart';
+
+abstract class HiveService {
+  Future<void> saveEthPendingTxAmount(
+      EthereumPendingTxAmount tx, String address);
+
+  Future<void> deleteEthPendingTxAmount(String txHash, String address);
+
+  Future<List<EthereumPendingTxAmount>> getEthPendingTxAmounts(String address);
+}
+
+class HiveServiceImpl implements HiveService {
+  static const ethPendingTxBox = 'ethPendingTxBox';
+
+  @override
+  Future<void> deleteEthPendingTxAmount(String txHash, String address) async {
+    final key = address.toLowerCase();
+    try {
+      final box = Hive.box<List<EthereumPendingTxAmount>>(ethPendingTxBox);
+      final List<EthereumPendingTxAmount> existingList =
+          box.get(key, defaultValue: null) ?? [];
+      if (existingList.any((element) => element.txHash == txHash)) {
+        existingList.removeWhere((element) => element.txHash == txHash);
+        await box.put(key, existingList);
+      }
+    } catch (e) {
+      log.info('Hive error deleting tx from Hive: $e');
+      return Future.value();
+    }
+  }
+
+  @override
+  Future<List<EthereumPendingTxAmount>> getEthPendingTxAmounts(
+      String address) async {
+    final key = address.toLowerCase();
+    try {
+      final box =
+          await Hive.openBox<List<EthereumPendingTxAmount>>(ethPendingTxBox);
+
+      final List<EthereumPendingTxAmount> existingList =
+          box.get(key, defaultValue: null) ?? [];
```

Comment:
`box.get(key, defaultValue: [])` Does this work?

Reply:
still have to check null